pkg/query-service/app/clickhouseReader/reader.go

```diff
@@ -1326,6 +1326,729 @@ func (r *ClickHouseReader) SearchSpans(ctx context.Context, queryParams *model.S
 	return &searchSpansResult, nil
 }
 
+func (r *ClickHouseReader) GetSpanFilters(ctx context.Context, queryParams *model.SpanFilterParams) (*model.SpanFiltersResponse, *model.ApiError) {
+
+	var query string
+	args := []interface{}{strconv.FormatInt(queryParams.Start.UnixNano(), 10), strconv.FormatInt(queryParams.End.UnixNano(), 10)}
+	if len(queryParams.ServiceName) > 0 {
+		for i, e := range queryParams.ServiceName {
+			if i == 0 && i == len(queryParams.ServiceName)-1 {
+				query += " AND (serviceName=?)"
+			} else if i == 0 && i != len(queryParams.ServiceName)-1 {
+				query += " AND (serviceName=?"
+			} else if i != 0 && i == len(queryParams.ServiceName)-1 {
+				query += " OR serviceName=?)"
+			} else {
+				query += " OR serviceName=?"
+			}
+			args = append(args, e)
+		}
+	}
+	if len(queryParams.HttpRoute) > 0 {
```

Comment:
Create more individual functions for conditional statements like switch case/if else statements